packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3730,11 +3730,8 @@ abstract class ComponentElement extends Element {
   /// [rebuild] when the element needs updating.
   @override
   void performRebuild() {
-    assert(() {
-      if (debugProfileBuildsEnabled)
-        Timeline.startSync('${widget.runtimeType}',  arguments: timelineWhitelistArguments);
-      return true;
-    }());
+    if (debugProfileBuildsEnabled)
```

Comment:
This also needs to be behind the kReleaseModeFlag

Reply:
Done.